src/serialbox-fortran/utils_ppser.f90

```diff
@@ -75,17 +76,17 @@ SUBROUTINE ppser_initialize(directory, prefix, mode, prefix_ref, mpi_rank, rprec
   IF ( .NOT. ppser_initialized ) THEN
     IF ( PRESENT(mpi_rank) ) THEN
       WRITE(suffix, '(A5,I0)') "_rank", mpi_rank
-      CALL fs_create_serializer(directory, TRIM(prefix)//TRIM(suffix), 'w', ppser_serializer)
+      CALL fs_create_serializer(directory, TRIM(prefix)//TRIM(suffix), 'w', ppser_serializer, opt_archive)
```

Comment:
Sorry, i think you are wrong. When an non-present optional argument is passed to another routine which also declares this argument as optional, the argument will also just be not present in the called routine. So, I'm pretty sure that you don't need an IF (PRESENT(opt_archive)) here, but haven't double-checked it.